src/main/modules/ardoq/YarnParser.ts

```diff
@@ -0,0 +1,28 @@
+import { HTTPError } from '../../HttpError';
+
+import { Dependency } from './Dependency';
+import { IParser } from './IParser';
+
+import yaml from 'js-yaml';
+
+export class YarnParser implements IParser {
+  public extractTopTierDeps(depString: string): Dependency[] {
+    const deps: Dependency[] = [];
+    yaml.loadAll(depString, function (dep: unknown) {
+      for (const [key, value] of Object.entries(dep as Record<string, YamlDependency>)) {
+        if (key !== '__metadata') {
+          if (value.version === undefined) {
+            throw new HTTPError(`Failed to parse yarn.lock file. Failed parsing key: ${key} value: ${value}`, 400);
```

Comment:
Minor niggle, the YarnParser probably should know about HTTPError it should just throw an error and then what ever is responsible for handling the HTTP connection should decide it's a 400. That'll probably add a bit of mess wrapping this call with a try catch and re-throwing with a HTTPError though so don't do it if it's a waste of time.

Also, do you want to return 400 if a single dep is broken? Is this based on a real world situation? My approach probably would have been:

```
return Object.entries(...)
  .filter(([key, value]) => key !== '__metadata' && value.version !== undefined))
  .map(([key, value]) => new Dependency(...))
```

Reply:
Ah I see you're building up a big list with deps... so probably

```
const fileDeps = Object.entries(...) 
  .filter(([key, value]) => key !== '__metadata' && value.version !== undefined)) 
  .map(([key, value]) => new Dependency(...))

deps.push(...fileDeps)
```

Key question though - is `.loadAll` async? Because I don't think this will work if it is

Reply:
I'll resolve the http errors and general approach for handling individual dep errors in a follow up PR.
I've changed this up to use the .load method instead which has tidied things a little at least